src/utils/resmoke/index.ts

```diff
@@ -54,4 +62,50 @@ const processResmokeLine = (line: string) => {
   return output;
 };
 
-export { processResmokeLine };
+/**
+ * `getColorMapping` returns a mapping of colors to resmoke functions
+ * @param logLine a log line from resmoke
+ * @param portColors a map of ports to colors for already seen ports
+ * @returns a color code for the port or undefined if the line is not a resmoke line
+ */
+const getColorMapping = (
+  logLine: string,
+  portColors: Record<string, string>
+) => {
+  if (!logLine) {
+    return;
+  }
+  const portOrState: string | undefined = getPort(logLine) ?? getState(logLine);
```

Comment:
You can remove the type here if you would like, the compiler is smart enough to figure it out